LibreHardwareMonitorLib/Hardware/Storage/StorageGroup.cs

```diff
@@ -27,6 +30,7 @@ public StorageGroup(ISettings settings)
             uint idx = Convert.ToUInt32(diskDrive.Properties["Index"].Value);
             ulong diskSize = Convert.ToUInt64(diskDrive.Properties["Size"].Value);
             int scsi = Convert.ToInt32(diskDrive.Properties["SCSIPort"].Value);
+            string serialNumber = ExtractSerialNumber((string)diskDrive.Properties["SerialNumber"].Value);
```

Comment:
I've mentioned this before in another PR, but a serial number is not a reliable way to map for HDDs, I have two HDDs here with the same serial number.

Reply:
I see MSFT_PhysicalDisk and Win32_DiskDrive both using a DeviceID. This is probably more reliable and possibly wouldn't require extracting the serial then, have you tried this?

Reply:
Yeah, I think device id is the way to go. The ObjectID might not be needed as you should be able to traverse through the WMI objects, see comment below. I'll mark this as resolved.